### ISSUE TITLE
Revert operation fails when reverted author is currently a type/redirect 

### ISSUE THREAD

**Issue by hornc** [internetarchive/openlibrary#475]
Tried reverting https://openlibrary.org/works/OL14998089W to v=2

but the old author is now  a redirect so the revert is failing.

```
Python	/opt/openlibrary/deploys/openlibrary/15dddfb/infogami/infobase/client.py in handle_error, line 84
Web	POST https://openlibrary.org/works/OL14998089W
```

```
Variable	Value   data	
{'at': {'key': '/works/OL14998089W', 'property': 'authors.author'}, 'error': 'bad_data', 'message': 'expected /type/author, found /type/redirect', 'value': '/authors/OL22105A'}
error	
'{"message": "expected /type/author, found /type/redirect", "at": {"property": "authors.author", "key": "/works/OL14998089W"}, "value": "/authors/OL22105A", "error": "bad_data"}'
json	
'{"message": "expected /type/author, found /type/redirect", "at": {"property": "authors.author", "key": "/works/OL14998089W"}, "value": "/authors/OL22105A", "error": "bad_data"}'
message	
'expected /type/author, found /type/redirect'
self	
<infogami.infobase.client.RemoteConnection instance at 0x7fa5b2012f80>
status	
'400 Bad Request'
```

**Comment by guyjeangilles** [comment]
Will probably be fixed by #2552. Assigning @hornc per slack discussions since his label is on this issue

**Comment by hornc** [comment]
specific item is fixed, fixing the general problem in #2552
